nmdc_schema/migrators/migrator_from_X_to_PR10.py

```diff
@@ -1,8 +1,22 @@
-from nmdc_schema.migrators.migrator_base import MigratorBase
 from typing import Optional
+
+from nmdc_schema.migrators.migrator_base import MigratorBase
+from nmdc_schema.nmdc_data import get_nmdc_schema_definition
 from linkml_runtime import SchemaView
 
 
+def create_schema_view() -> SchemaView:
+    """
+    Returns a LinkML SchemaView instance that can be used to traverse the schema.
+    
+    >>> isinstance(create_schema_view(), SchemaView)
+    True
```

Comment:
This doctest is currently failing (see GitHub Actions workflow logs).

Reply:
One of the steps that precedes the doctest, deletes that file: https://github.com/microbiomedata/berkeley-schema-fy24/actions/runs/9073086951/job/24930084750#step:7:17 🤔

Reply:
Hi @turbomam, I wrote a test today that targets a function that calls `get_nmdc_schema_definition()` (imported from `nmdc_schema.nmdc_data`), which depends upon the existence of a file at `nmdc_schema/nmdc_schema_merged.yaml`. That file does exist in this repo.

The test is failing in GitHub Actions because the `make test` command (which is how tests are being run in GitHub Actions) deletes that file before running the doctests.

Do you have any suggestions as to how I resolve this? For example:
- A: Updating the `make test` Makefile target so that the file doesn't get deleted before the doctests run?
- B: Document a limitation of the repo saying that doctests cannot do things that depend upon the existence of the files that the `make test` Makefile target deletes?

Reply:
Nice detective work. 

I'm inclined to update the `test` target, unless yo think you have a high quality, non-make way of running the tests in the action. Maybe that would just be running pytest in discovery mode? I'd like for us to touch base with @cmungall @sierra-moxon or @pkalita-lbl to see if nmdc-schema is even using the LinkML-preferred test harness. I'm not even sure why I was deleting `nmdc_schema/nmdc_schema_merged.yaml`. It could be obsessive tidiness or there could be a good reason I just can't remember now.

The new `create_schema_view()` and modified `Migrator(MigratorBase)` are fine with me. Do they resolve the problem you are experiencing?

Reply:
>Do they resolve the problem you are experiencing?

Yes, they do. I confirmed that — with those changes in place on this branch — I can run the migrator when importing this branch as a package (installed via `pip install git+https://github.com/microbiomedata/berkeley-schema-fy24@base-issue-1970-migrator-fails-to-load-YAML-file-when-used-via-PyPI-package`).

---

Regarding updating the `test` target:

Here's the chain of Makefile targets that currently causes the doctests to run. They are run via the `site` target (it looks to me like they are the only tests that are run that way). Note that there are other Makefile targets run via `make test` — this diagram just shows the two involved in this issue.

```mermaid
graph LR
  test --> site
  site --> site-clean
  site-clean -..- site-clean-commands(["
	rm -rf nmdc_schema/*.json
	rm -rf nmdc_schema/*.tsv
	rm -rf nmdc_schema/*.yaml
  "])
  site --> migration-doctests
  migration-doctests -..- commands(["
	$(RUN) python -m doctest -v nmdc_schema/migrators/*.py
	$(RUN) python -m doctest -v nmdc_schema/migrators/adapters/*.py
	$(RUN) python -m doctest -v nmdc_schema/migrators/cli/*.py
  "])
```

I'll update this branch with a proposal for running the doctests in a way that works around this issue.

Reply:
Looks like `make site` eventually creates that YAML file, but it comes later in the sequence of dependencies.

![image](https://github.com/microbiomedata/berkeley-schema-fy24/assets/134325062/4d685e6b-34c2-4f61-8dea-6aa221a61cf1)

One approach is to reorder the dependencies of the `site` target so that the doctests are only run after those files get created. I opted to, instead, remove the doctest-running target from the dependency list of `site` _entirely_ and add it to the dependency list of `test`, which makes more sense to me as a home for it (more of a separation of concerns, in my opinion).

Reply:
was just experimenting with adding `project/nmdc_schema_merged.yaml` after `gen-project` so I +1 your approach :)